src/torchcodec/samplers/_implem.py

```diff
@@ -160,12 +160,17 @@ def _build_all_clips_indices(
 def _decode_all_clips_indices(
     decoder: VideoDecoder, all_clips_indices: list[int], num_frames_per_clip: int
 ) -> list[FrameBatch]:
-    # This takes the list of all the frames to decode, decode all the frames,
-    # and then packs them into clips of length num_frames_per_clip.
-    # This is slow, unoptimized, and u.g.l.y. It is not meant to stay.
-    # TODO:
-    # - sort the frames to avoid backward seeks, dedup, decode, and re-organize frames.
-    # - write most of this in C++
+    # This takes the list of all the frames to decode (in arbitrary order),
+    # decode all the frames, and then packs them into clips of length
+    # num_frames_per_clip.
+    #
+    # To avoid backwards seeks (which are slow), we:
+    # - sort all the frame indices to be decoded
+    # - dedup them
+    # - decode all unique frames in sorted order
+    # - re-assemble the decoded frames back to their original order
+    #
+    # TODO: Write this in C++ so we can avoid the copies that happen in `to_framebatch`
```

Comment:
Apologies for the use of  `i` and `j` indices. Kinda hard to find good names for "indices within an array of indices"